src/transactions.rs

```diff
@@ -0,0 +1,456 @@
+    /// finalized.
+    ///
+    /// For native transfers, the transfer inputs and the fee input get mixed together, so there is
+    /// only one change output, which accounts for both the fee change and the transfer change. In
+    /// this case, the total amount of change will be reflected in `fee_change` and `asset_change`
+    /// will be `Some(0)`.
+    ///
+    /// Note that `None` indicates that the amount of change is unknown, not that there is no
+    /// change, which would be indicated by `Some(0)`. The amount of change may be unknown if, for
+    /// example, this is a transaction we received from someone else, and we do not hold the
+    /// necessary viewing keys to inspect the change outputs of the transaction.
+    asset_change: Option<RecordAmount>,
+    /// If we sent this transaction, a receipt to track its progress.
+    receipt: Option<TransactionReceipt<L>>,
+}
+
+impl<L: Ledger> Transaction<L> {
+    #![allow(dead_code)]
+    pub fn uid(&self) -> &TransactionUID<L> {
+        &self.uid
+    }
+    pub fn timeout(&self) -> u64 {
+        self.timeout
+    }
+    pub fn memos(&self) -> &Vec<Option<ReceiverMemo>> {
+        &self.memos
+    }
+    pub fn accounts(&self) -> &Vec<UserAddress> {
+        &self.accounts
+    }
+    pub fn sig(&self) -> &Signature {
+        &self.sig
+    }
+    pub fn freeze_outputs(&self) -> &Vec<RecordOpening> {
+        &self.freeze_outputs
+    }
+    pub fn inputs(&self) -> &Vec<RecordOpening> {
+        &self.inputs
+    }
+    pub fn outputs(&self) -> &Vec<RecordOpening> {
+        &self.outputs
+    }
+    pub fn time(&self) -> &DateTime<Local> {
+        &self.time
+    }
+    pub fn asset(&self) -> &AssetCode {
+        &self.asset
+    }
+    pub fn kind(&self) -> &TransactionKind<L> {
+        &self.kind
+    }
+    pub fn senders(&self) -> &Vec<UserAddress> {
+        &self.senders
+    }
+    pub fn receivers(&self) -> &Vec<(UserAddress, RecordAmount)> {
+        &self.receivers
+    }
+    pub fn fee_change(&self) -> &Option<RecordAmount> {
+        &self.fee_change
+    }
+    pub fn asset_change(&self) -> &Option<RecordAmount> {
+        &self.asset_change
+    }
+    pub fn receipt(&self) -> &Option<TransactionReceipt<L>> {
+        &self.receipt
+    }
+}
+
+type TransactionsStore<L> = KeyValueStore<TransactionUID<L>, Transaction<L>>;
+
+/// An editor to create or update the transaction or transactions store.
+pub struct TransactionEditor<'a, L: Ledger + Serialize + DeserializeOwned> {
+    transaction: Transaction<L>,
+    store: &'a mut TransactionsStore<L>,
+}
+
+impl<'a, L: Ledger + Serialize + DeserializeOwned> TransactionEditor<'a, L> {
+    /// Create a transaction editor.
+    fn new(store: &'a mut TransactionsStore<L>, transaction: Transaction<L>) -> Self {
+        Self { transaction, store }
+    }
+
+    /// Add fee change record to the transaction once it is certain
+    pub fn with_fee_change(mut self, amount: RecordAmount) -> Self {
```

Comment:
Would it be useful to add `set_*` and `clear_*` functions (e.g., `set_fee_change` and `clear_fee_change`), similar to `set_name` and `clear_name` in `AssetEditor`? I found them helpful for asset operations, but not sure if that's the case for transactions.

Reply:
The reason I left them out is because all of the fields with only a `with_*` setter should be set to a value only once and then remain the same.  I think adding `set_*` and `clear_*` would be confusing if they should never be called but maybe there is a reason I am not thinking of.